util/gnmi.go

```diff
@@ -92,6 +97,31 @@ func PathMatchesPathElemPrefix(path, prefix *gpb.Path) bool {
 	return true
 }
 
+// PathMatchesQuery returns whether query is prefix of path.
+// Only the query may contain wildcard name or keys.
+// TODO: Multilevel wildcards ("...") not supported.
+// Both paths must use the gNMI >=0.4.0 PathElem path format.
+func PathMatchesQuery(path, query *gpb.Path) bool {
+	if len(path.GetElem()) < len(query.GetElem()) || path.Origin != query.Origin {
+		return false
+	}
+	for i, queryElem := range query.Elem {
+		pathElem := path.Elem[i]
+		if queryElem == nil || pathElem == nil {
+			return false
+		}
```

Comment:
Could you document this corner case behaviour?

Reply:
adde to godoc